bin/xlsx2json.py

```diff
@@ -208,9 +210,16 @@ def get_project_json_data(self):
     def get_biosample_object(self, data):
         sample_name = self.xlsx_conf[SAMPLE][OPTIONAL_HEADERS_KEY_NAME][SAMPLE_NAME_KEY]
         scientific_name = self.xlsx_conf[SAMPLE][OPTIONAL_HEADERS_KEY_NAME][SCIENTIFIC_NAME_KEY]
-        if sample_name not in data or scientific_name not in data:
-            raise ValueError(f'If BioSample Accession is not provided, '
-                             f'The {SAMPLE} worksheet should have {SAMPLE_NAME_KEY} and {SCIENTIFIC_NAME_KEY} populated')
+        if sample_name not in data:
+            self.add_error(f'If BioSample Accession is not provided, the {SAMPLE} worksheet should have '
+                           f'{SAMPLE_NAME_KEY} populated',
+                           sheet=SAMPLE, column=SAMPLE_NAME_KEY)
+            return
+        if scientific_name not in data:
+            self.add_error(f'If BioSample Accession is not provided, the {SAMPLE} worksheet should have '
+                           f'{SCIENTIFIC_NAME_KEY} populated',
+                           sheet=SAMPLE, column=SCIENTIFIC_NAME_KEY)
+            return
```

Comment:
I think it would make sense to make a list of mandatory sample name column and iterate over each field to add errors. You can then return at the end.

Reply:
My thought was doing the return here mimics raising the exception as closely as possible, so I wouldn't have to change the surrounding logic as much (for example the rest of this method assumes these fields are present).

But you raise a good point in that I think this won't work without also checking for errors outside this method, and so on... let me think about it again...

Reply:
I think it's better now, won't get one error for each row:
![Screenshot from 2024-04-30 16-03-37](https://github.com/EBIvariation/eva-sub-cli/assets/5599307/9b5c1d0b-218f-4064-af3f-d4f3f5e9c929)